### PR TITLE
Notebook validation warning logs

### PR DESCRIPTION
**Pull request by PrajwalPrashanth**
Dependent of backend PR: 468

image
image

Works for colab notebooks too
image

### CODE REVIEW

File: jovian/utils/api.py [hunk 1]
```diff
@@ -76,6 +76,8 @@ def create_gist_simple(filename=None, gist_slug=None, privacy='auto', title=None
                        files={'files': nb_file},
                        headers=auth_headers)
             if res.status_code == 200:
+                if res.json().get('errors'):
```
Comment (sidujjain) [inline jovian/utils/api.py:79]: seems like you can add a variable for res.json()

File: jovian/utils/api.py [hunk 2]
```diff
@@ -132,3 +140,9 @@ def post_slack_message(data, safe=False):
         return {'data': {'messageSent': False}}
     else:
         raise ApiError('Slack trigger failed: ' + pretty(res))
+
+
+def parse_warning(result):
```
Comment (sidujjain) [inline jovian/utils/api.py:145]: you can modify it to `parse_result(res)` and return `data, error` tuple

**Comment by aakashns** [comment]
Please implement it in a backward-compatible fashion, so that it works for the old and the new backend. Also, there seems to be an unresolved comment, please resolve.

File: jovian/tests/utils/test_api.py [hunk 3]
```diff
@@ -338,6 +340,8 @@ def test_create_gist_simple_no_gist_slug(mock_requests_post):
             version_title="first version",
         )
 
+        assert capsys.readouterr().err.strip() == '[jovian] Error: Uploaded gist has a warning'.strip()
```
Comment (sidujjain) [inline jovian/tests/utils/test_api.py:343]: Instead of checking for the warning in all the tests, it would be better to create a specific test for that.

**Review by sidujjain — APPROVED** [review]

**Pull request merged**